src/main/java/seedu/address/model/Model.java

```diff
@@ -112,4 +113,10 @@ public interface Model {
      * Clears the note panel display.
      */
     void clearDisplayNote();
+
+    /**
+     * Checks if the person with the inputted ic number exists in ClinicMate.
+     * Returns the person if they exist.
+     */
+    Person getPersonIfExists(IdentityCardNumberMatchesPredicate icPredicate) throws CommandException;
```

Comment:
The CommandException should not be thrown at this point. It should be thrown by the Command itself (since it's a **Command** Exception). If it were me, I would do it by making the method:

`Optional<Person> getPersonIfExists(IdentityCardNumberMatchesPredicate icPredicate)`.

Reply:
Then in the commands itself, you could do something like

`Person person = model.getPersonIfExists(icPredicate).orElseThrow(() -> new CommandException(Messages.MESSAGE_NO_MATCHING_IC_DELETE));`

This also has the added benefit that the different commands can throw different messages (`MESSAGE_NO_MATCHING_IC` vs `MESSAGE_NO_MATCHING_IC_DELETE`)